scripts/install_avalanchego_release.sh

```diff
@@ -27,22 +27,72 @@ if [[ ${GOOS} == "darwin" ]]; then
   AVAGO_DOWNLOAD_PATH=${BASEDIR}/avalanchego-macos-${VERSION}.zip
 fi
 
-AVALANCHEGO_BUILD_PATH=${AVALANCHEGO_BUILD_PATH:-${BASEDIR}/avalanchego-${VERSION}}
+# TODO: Please read:
+# not sure why the version isn't included in CI currently... open to suggestions on how to handle this better
+AVALANCHEGO_BUILD_PATH=${AVALANCHEGO_BUILD_PATH:-${BASEDIR}/avalanchego}
+VERSIONED_AVALANCHEGO_BUILD_PATH=${AVALANCHEGO_BUILD_PATH}-${VERSION}
 mkdir -p $AVALANCHEGO_BUILD_PATH
+mkdir -p $VERSIONED_AVALANCHEGO_BUILD_PATH
```

Comment:
I agree with this, I'd prefer to just use a single `AVALANCHEGO_BUILD_PATH` and either build in that location or unpack the downloaded release into that location rather than adding a separate versioned path.

Reply:
@aaronbuchwald, why do you prefer a single build path? 

@ceyonur, the reason I did this was actually to avoid further nesting of logic, making the whole script harder to read. Not the best reason, but a reason, none-the-less.

I'm now thinking that `AVALANCHE_VERSION` is too overloaded and that it would be better to separate things into `AVALANCHE_RELEASE_VERSION`, `AVALANCHE_COMMIT`, and `AVALANCHE_BRANCH` and the logic for each variant is completely different (branches need to pulled in every time).

@aaronbuchwald, I also think it makes a lot more sense to have versioned paths, the same way that `cargo` stores dependencies for instance. I can make changes such that we don't rebuild if an existing commit or release-tagged version already exists, and only rebuild for a specified branch. 

Thoughts?

Reply:
Sounds good to me 👍